samples/samples/pg_snippets.py

```diff
@@ -0,0 +1,1542 @@
+#!/usr/bin/env python
+
+# Copyright 2022 Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This application demonstrates how to do basic operations using Cloud
+Spanner PostgreSql dialect.
+
+For more information, see the README.rst under /spanner.
+"""
+import argparse
+import base64
+import datetime
+import decimal
+import time
+
+from google.cloud import spanner, spanner_admin_database_v1
+from google.cloud.spanner_admin_database_v1.types.common import DatabaseDialect
+from google.cloud.spanner_v1 import param_types
+
+OPERATION_TIMEOUT_SECONDS = 240
+
+
+# [START spanner_postgresql_create_instance]
+def create_instance(instance_id):
+    """Creates an instance."""
+    spanner_client = spanner.Client()
+
+    config_name = "{}/instanceConfigs/regional-us-central1".format(
+      spanner_client.project_name
+    )
+
+    instance = spanner_client.instance(
+      instance_id,
+      configuration_name=config_name,
+      display_name="This is a display name.",
+      node_count=1,
+      labels={
+        "cloud_spanner_samples": "true",
+        "sample_name": "snippets-create_instance-explicit",
+        "created": str(int(time.time())),
+      },
+    )
+
+    operation = instance.create()
+
+    print("Waiting for operation to complete...")
+    operation.result(OPERATION_TIMEOUT_SECONDS)
+
+    print("Created instance {}".format(instance_id))
+
+
+# [END spanner_postgresql_create_instance]
+
+
+# [START spanner_postgresql_create_database]
+def create_database(instance_id, database_id):
+    """Creates a PostgreSql database and tables for sample data."""
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+
+    database = instance.database(
+      database_id,
+      database_dialect=DatabaseDialect.POSTGRESQL,
+    )
+
+    operation = database.create()
+
+    print("Waiting for operation to complete...")
+    operation.result(OPERATION_TIMEOUT_SECONDS)
+
+    create_table_using_ddl(database.name)
+    print("Created database {} on instance {}".format(database_id, instance_id))
+
+
+def create_table_using_ddl(database_name):
+    spanner_client = spanner.Client()
+    request = spanner_admin_database_v1.UpdateDatabaseDdlRequest(
+      database=database_name,
+      statements=[
+        """CREATE TABLE Singers (
+  SingerId   bigint NOT NULL,
```

Comment:
nit: can you format this?

Reply:
Also can you adda comment that pg is case insensitive or rename it to lower case